core/ordersync/ordersync.go

```diff
@@ -312,65 +311,60 @@ func (s *Service) GetOrders(ctx context.Context, minPeers int) error {
 				"provider": peerID.Pretty(),
 			}).Trace("requesting orders from neighbor via ordersync")
 
+			select {
```

Comment:
I think this `select` statement belongs at the beginning of the `for _, peerID := range currentNeighbors` loop rather than here. We don't want to check `len(successfullySyncedPeers)` or log `"requesting orders from neighbor via ordersync"` until _after_ a slot opens up in the `semaphore` channel.